src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,693 @@
+    /// token0 is also compatible with the Uniswap V2 Router interface.
+    IERC20 public immutable token0;
+    /// @notice The swap output token that is transferred from this contract.
+    /// From a User perspective, this is the token being bought.
+    /// token1 is also compatible with the Uniswap V2 Router interface.
+    IERC20 public immutable token1;
+
+    ////////////////////////////////////////////////////
+    ///             Storage Variables
+    ////////////////////////////////////////////////////
+
+    /**
+     * @notice For one `token0` from a Trader, how many `token1` does the pool send.
+     * For example, if `token0` is WETH and `token1` is stETH then
+     * `traderate0` is the WETH/stETH price.
+     * From a Trader's perspective, this is the buy price.
+     * From the ARM's perspective, this is the sell price.
+     * Rate is to 36 decimals (1e36).
+     * To convert to a stETH/WETH price, use `PRICE_SCALE * PRICE_SCALE / traderate0`.
+     */
+    uint256 public traderate0;
+    /**
+     * @notice For one `token1` from a Trader, how many `token0` does the pool send.
+     * For example, if `token0` is WETH and `token1` is stETH then
+     * `traderate1` is the stETH/WETH price.
+     * From a Trader's perspective, this is the sell price.
+     * From a ARM's perspective, this is the buy price.
+     * Rate is to 36 decimals (1e36).
+     */
+    uint256 public traderate1;
+
+    /// @notice cumulative total of all withdrawal requests included the ones that have already been claimed
+    uint128 public withdrawsQueued;
+    /// @notice total of all the withdrawal requests that have been claimed
+    uint128 public withdrawsClaimed;
+    /// @notice cumulative total of all the withdrawal requests that can be claimed including the ones already claimed
+    uint128 public withdrawsClaimable;
+    /// @notice index of the next withdrawal request starting at 0
+    uint128 public nextWithdrawalIndex;
+
+    struct WithdrawalRequest {
+        address withdrawer;
+        bool claimed;
+        // When the withdrawal can be claimed
+        uint40 claimTimestamp;
+        // Amount of assets to withdraw
+        uint128 assets;
+        // cumulative total of all withdrawal requests including this one.
+        // this request can be claimed when this queued amount is less than or equal to the queue's claimable amount.
+        uint128 queued;
+    }
+
+    /// @notice Mapping of withdrawal request indices to the user withdrawal request data
+    mapping(uint256 requestId => WithdrawalRequest) public withdrawalRequests;
+
+    /// @notice The account that can collect the performance fee
+    address public feeCollector;
+    /// @notice Performance fee that is collected by the feeCollector measured in basis points (1/100th of a percent)
+    /// 10,000 = 100% performance fee
+    /// 2,000 = 20% performance fee
+    /// 500 = 5% performance fee
+    uint16 public fee;
```

Comment:
⛽️ So the intention here in the storage sizes is that `fee`, `feesAccrued`, `lastAvailableAssets` will fit into the same storage slot. The three of these are read together, and two of these are written together.

HOWEVER, this is not actually happening under the hood. While all three fit into a single storage slot, the compiler greedy matching, and is looking at:

```solidity
address public feeCollector;
uint16 public fee
uint112 public feesAccrued;
uint128 public lastAvailableAssets;
```

And in it's greedy little packing brain is doing:

```
[address, uint16] + [uint112, uint128]
```

Fortunately, we don't write to the `fee` variable, so we aren't paying an extra write, but it does add an extra 2,000 gas cold storage read to every time we use this variable set.

I've tested reordering it to

```solidity
    uint128 public lastAvailableAssets;
    uint112 public feesAccrued;
    uint16 public fee;
    
    address public feeCollector;
```

Which saves 1,600 to 2,000 gas in the relevant forge tests. Swaps are unaffected.

<img width="1161" alt="image" src="https://github.com/user-attachments/assets/ad34a428-890d-4c73-9913-d457179a3093">

Reply:
Good spot. I've reordered and checked the slots using 
```
forge inspect LidoARM storage
```

Reply:
`feesAccrued` has been removed so the fee variables are now

```
uint16 public fee;
int128 public lastAvailableAssets;
address public feeCollector;
```